plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -294,6 +324,52 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
                 ImmutableList.of());
     }
 
+    @Override
+    public boolean isSupportedVersionType(ConnectorSession session, SchemaTableName tableName, PointerType pointerType, io.trino.spi.type.Type versioning)
+    {
+        switch (pointerType) {
+            case TEMPORAL:
+                return versioning instanceof TimestampWithTimeZoneType || versioning instanceof TimestampType;
+            case TARGET_ID:
+                return versioning == BIGINT;
+        }
+        return false;
+    }
+
+    private long getSnapshotIdFromVersion(Table table, ConnectorTableVersion version)
+    {
+        io.trino.spi.type.Type versionType = version.getVersionType();
+        switch (version.getPointerType()) {
+            case TEMPORAL:
+                verify(versionType instanceof TimestampWithTimeZoneType || versionType instanceof TimestampType,
+                        "Iceberg temporal time travel only supports types TimestampWithTimeZone and Timestamp, but got " + versionType);
+
+                long epochMillis;
+                if (versionType instanceof TimestampWithTimeZoneType) {
+                    if (((TimestampWithTimeZoneType) versionType).isShort()) {
+                        epochMillis = unpackMillisUtc((long) version.getVersion());
+                    }
+                    else {
+                        LongTimestampWithTimeZone timestamp = (LongTimestampWithTimeZone) version.getVersion();
+                        epochMillis = timestamp.getEpochMillis();
+                    }
+                }
+                else {
+                    if (((TimestampType) versionType).isShort()) {
+                        epochMillis = ((long) version.getVersion()) / MICROSECONDS_PER_MILLISECOND;
+                    }
+                    else {
+                        epochMillis = ((LongTimestamp) version.getVersion()).getEpochMicros() / MICROSECONDS_PER_MILLISECOND;
+                    }
+                }
+                return getSnapshotIdAsOfTime(table, epochMillis);
+            case TARGET_ID:
+                verify(versionType == BIGINT, "Iceberg target ID time travel only supports type Bigint, but got " + versionType);
+                return (long) version.getVersion();
+        }
+        throw new TrinoException(NOT_SUPPORTED, "Version pointer type " + version.getPointerType() + " is not supported");
```

Comment:
use format, but even better, move variablel part to the end:

```suggestion
        throw new TrinoException(NOT_SUPPORTED, "Version pointer type is not supported: " +  version.getPointerType());
```